pkg/controller/handler.go

```diff
@@ -113,13 +113,17 @@ func (c *Controller) handleVirtualService(ingress *networkingv1.Ingress) error {
 		if err != nil {
 			return err
 		}
-	} else if !reflect.DeepEqual(vs.Spec, nvs.Spec) {
-		klog.Infof("updated virtual service \"%s/%s\"", vs.Namespace, vs.Name)
+	} else if !reflect.DeepEqual(vs.ObjectMeta.Labels, nvs.ObjectMeta.Labels) || !reflect.DeepEqual(vs.ObjectMeta.Annotations, nvs.ObjectMeta.Annotations) || !reflect.DeepEqual(vs.Spec, nvs.Spec) {
```

Comment:
Hmmm... What happens if there are other labels added to this from somewhere else? Would it be better to only target specific labels in the comparison?

Reply:
That would have caused an issue, so that's why down in `generateObjectMetadata` it copies all the existing ones first and then edits the ones we want to maintain!

Reply:
Oh I see now! Thanks!